endpoints.tf

```diff
@@ -18,7 +18,7 @@ resource "aws_vpc_endpoint" "transfer_server" {
 # Resources for the S3 VPC service endpoint
 resource "aws_vpc_endpoint" "s3" {
   count             = var.private_s3_endpoint ? 1 : 0
-  route_table_ids   = aws_route_table.private[*].id
+  route_table_ids   = var.s3_endpoint.route_table_ids
```

Comment:
I think you should use some logic here that uses either the provided s3 endpoint or falls back to the value it had previously...

Reply:
The logic for the ec2 endpoints was added like that initially, This now already exists and with this change your making it backwards incompatible.

Reply:
@svanharmelen will conditional expression work for this such as below ? 

`route_table_ids   = var.s3_endpoint.route_table_ids != "" ? var.s3_endpoint.route_table_ids : "aws_route_table.private[*].id"`

Reply:
```
locals {
  s3_route_table_ids = var.s3_endpoint.route_table_ids != null ? var.s3_endpoint.route_table_ids : aws_route_table.private[*].id
}
```
And then:
```
route_table_ids = local.s3_route_table_ids
```

Reply:
thats in as local now